packages/opentelemetry-exporter-zipkin/src/zipkin.ts

```diff
@@ -68,6 +69,17 @@ export class ZipkinExporter implements SpanExporter {
     spans: ReadableSpan[],
     resultCallback: (result: ExportResult) => void
   ) {
+    if (typeof this._serviceName !== 'string') {
```

Comment:
This whole if tree could be drastically simplified.

```typescript
if (typeof this._serviceName !== 'string') {
  this._serviceName = String(spans[0].resource.labels[SERVICE_RESOURCE.NAME] || 'Unnamed Service');
}
```

Reply:
The problem here is that the resource `service.name` can be either string, number or boolean. I needed to call `toString()` on that, so I had to check for it being defined.